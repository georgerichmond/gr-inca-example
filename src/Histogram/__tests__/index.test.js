import React from 'react';
import renderer from 'react-test-renderer';
import wait from 'waait';
import { MockedProvider } from 'react-apollo/test-utils';
import * as dateMock from 'jest-date-mock';
import resultBody from './result';
import { HistogramWithoutClientState as Histogram } from '../index';

import {
  query,
  edgesPath,
  edgeMetricTotalPath
} from '../../UniqueVisitorsHistogram';

jest.mock('../presentation', () => ({
  __esModule: true,
  HistogramPresentation: jest.fn(props => (
    <div id="HistogramPresentation">{JSON.stringify(props)}</div>
  )),
  HistogramLoading: jest.fn(props => (
    <div id="HistogramLoading">{JSON.stringify(props)}</div>
  ))
}));

describe('Histogram', () => {
  beforeAll(() => {
    dateMock.advanceTo(new Date('2018-09-19T22:00:00Z'));
  });

  afterAll(() => {
    dateMock.clear();
  });

  const activityPeriod = {
    from: '2018-09-19T00:00:00Z',
    to: '2018-09-19T23:00:00Z'
  };

  const articleId = 'asdfghjkl123456789';

  const interval = '1h';

  const name = 'Unique visitors';
  const metric = 'uniqueVisitors';

  it('renders just the label when the request is in flight', () => {
    const result = renderer.create(
      <MockedProvider mocks={[]} addTypename={false}>
        <Histogram
          activityPeriod={activityPeriod}
          articleId={articleId}
          edgesPath={edgesPath}
          edgeMetricTotalPath={edgeMetricTotalPath}
          query={query}
          name={name}
          metric={metric}
        />
      </MockedProvider>
    );

    expect(result).toMatchSnapshot();
  });

  it('renders ERROR when the request failed', async () => {
    const mocks = [
      {
        request: {
          query,
          variables: {
            interval,
            articleId,
            activityPeriod
          }
        },
        error: new Error('Not a Proper Request')
      }
    ];

    const result = renderer.create(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Histogram
          activityPeriod={activityPeriod}
          articleId={articleId}
          edgesPath={edgesPath}
          edgeMetricTotalPath={edgeMetricTotalPath}
          query={query}
          name={name}
          metric={metric}
        />
      </MockedProvider>
    );

    await wait(0);

    expect(result).toMatchSnapshot();
  });

  it('renders the histogram presentation when the request is successful', async () => {
    const mocks = [
      {
        request: {
          query,
          variables: {
            interval,
            articleId,
            activityPeriod
          }
        },
        result: resultBody
      }
    ];

    const result = renderer.create(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Histogram
          activityPeriod={activityPeriod}
          articleId={articleId}
          edgesPath={edgesPath}
          edgeMetricTotalPath={edgeMetricTotalPath}
          query={query}
          name={name}
          metric={metric}
        />
      </MockedProvider>
    );

    await wait(0);

    expect(result).toMatchSnapshot();
  });
});
