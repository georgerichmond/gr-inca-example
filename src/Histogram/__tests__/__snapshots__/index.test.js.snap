// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Histogram renders ERROR when the request failed 1`] = `
<div
  className="ant-alert ant-alert-warning"
  data-show={true}
>
  <i
    className="anticon anticon-exclamation-circle ant-alert-icon"
  >
    <svg
      aria-hidden="true"
      className=""
      data-icon="exclamation-circle"
      fill="currentColor"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm-32 232c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V296zm32 440a48.01 48.01 0 0 1 0-96 48.01 48.01 0 0 1 0 96z"
      />
    </svg>
  </i>
  <span
    className="ant-alert-message"
  >
    No chart data
  </span>
  <span
    className="ant-alert-description"
  />
</div>
`;

exports[`Histogram renders just the label when the request is in flight 1`] = `
<div
  id="HistogramLoading"
>
  {}
</div>
`;

exports[`Histogram renders the histogram presentation when the request is successful 1`] = `
<div
  id="HistogramPresentation"
>
  {"data":[{"x":"2018-09-30T00:00:00Z","y":9903},{"x":"2018-09-30T01:00:00Z","y":2506}],"name":"Unique visitors","metric":"uniqueVisitors","activityPeriod":{"from":"2018-09-19T00:00:00Z","to":"2018-09-19T23:00:00Z"}}
</div>
`;
